android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonsManagerAdapterTest.kt

```diff
@@ -572,4 +575,101 @@ class AddonsManagerAdapterTest {
         assertFalse(itemsWithSections.isEmpty())
         assertEquals(AddonsManagerAdapter.FooterSection, itemsWithSections.last())
     }
+
+    @Test
+    fun `bind blocklisted add-on`() {
+        Locale.setDefault(Locale.ENGLISH)
+        val addonsManagerAdapterDelegate: AddonsManagerAdapterDelegate = mock()
+        val titleView: TextView = mock()
+        whenever(titleView.context).thenReturn(testContext)
+        val summaryView: TextView = mock()
+        whenever(summaryView.context).thenReturn(testContext)
+        val statusBlocklistedView: View = mock()
+        val messageTextView: TextView = mock()
+        val learnMoreTextView = TextView(testContext)
+        whenever(statusBlocklistedView.findViewById<TextView>(R.id.add_on_status_blocklisted_message)).thenReturn(
+            messageTextView,
+        )
+        whenever(statusBlocklistedView.findViewById<TextView>(R.id.add_on_status_blocklisted_learn_more_link)).thenReturn(
+            learnMoreTextView,
+        )
+        val addonViewHolder = CustomViewHolder.AddonViewHolder(
+            view = View(testContext),
+            iconView = mock(),
+            titleView = titleView,
+            summaryView = summaryView,
+            ratingView = mock(),
+            ratingAccessibleView = mock(),
+            userCountView = mock(),
+            addButton = mock(),
+            allowedInPrivateBrowsingLabel = mock(),
+            statusBlocklistedView = statusBlocklistedView,
+        )
+        val addonName = "some addon name"
+        val addon = makeBlocklistedAddon(addonName)
+        val adapter = AddonsManagerAdapter(mock(), addonsManagerAdapterDelegate, emptyList())
+
+        adapter.bindAddon(addonViewHolder, addon)
+
+        verify(statusBlocklistedView).isVisible = true
+        verify(messageTextView).text = "$addonName has been disabled due to security or stability issues."
+
+        // Verify that a click on the "learn more" link actually does something.
+        learnMoreTextView.performClick()
+        verify(addonsManagerAdapterDelegate).onLearnMoreLinkClicked(
+            AddonsManagerAdapterDelegate.LearnMoreLinks.BLOCKLISTED_ADDON,
+            addon,
+        )
+    }
+
+    @Test
+    fun `bind blocklisted add-on without an add-on name`() {
+        Locale.setDefault(Locale.ENGLISH)
```

Comment:
As we are altering a global shared state between tests, maybe we should move this to the ```@Before setup``` section to make it more intentional that this will be shared between tests. If the intention is not to set this value for all tests, I think we should capture the current local at the beginning of the test and then set it back at the end.